### PR TITLE
Fix: Abnormal strings appear when comments are saved after editing

### PR DESCRIPTION
**Pull request by HEREYUA**
Partially resolved（The second problem）：   [#29986](https://github.com/go-gitea/gitea/issues/29986)

**Before**
HTML strings appear when comments are saved after editing

image

**After**

https://github.com/go-gitea/gitea/assets/37935145/525601f9-3ee1-4266-9105-36d82b91b1c8

### CODE REVIEW

**Review by yardenshoham — APPROVED** [review]

**Comment by lunny** [comment]
image
I don't know whether it's realted.

**Comment by yardenshoham** [comment]
I don't think so, how to reproduce?

**Comment by lunny** [comment]
I cannot reproduce it any more.

**Review by lunny — APPROVED** [review]

**Comment by silverwind** [comment]
> image I don't know whether it's realted.

This is either `web_src/js/features/repo-legacy.js:354` or `web_src/js/features/repo-legacy.js` and likely a separate issue.

File: web_src/js/features/repo-legacy.js [hunk 1]
```diff
@@ -436,7 +436,7 @@ async function onEditContent(event) {
       const $content = $segment;
       if (!$content.find('.dropzone-attachments').length) {
         if (data.attachments !== '') {
-          $content[0].append(data.attachments);
+          $content[0].insertAdjacentHTML('beforeend', data.attachments);
```
Comment (silverwind) [inline web_src/js/features/repo-legacy.js:439]: [Previous code](https://github.com/go-gitea/gitea/pull/29843/files#diff-4f89ee029f289a1c40cf85fa40b0a10046a93d5fe1dac8d158e98bebd2e6e49eL439-R439) had a `.dropzone-attachments` class, it it no longer needed to be created, @yardenshoham?

**Comment by silverwind** [comment]
Merged too fast...

File: web_src/js/features/repo-legacy.js (hunk 1 above)
Comment (yardenshoham) [reply web_src/js/features/repo-legacy.js:439]: It's immediately replaced...

Comment (silverwind) [reply web_src/js/features/repo-legacy.js:439]: Oh I see the previous code created and appended it, but then immediately replaced it with the HTML string, so likely it's okay.

**Review by silverwind — APPROVED** [review]

**Pull request merged**